In [ ]:
#First importa all the libraries that we need
import pandas as pd
import numpy as np
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [ ]:
#Read all the files that we need
gdp = pd.read_csv('gdp_csv.csv')
population = pd.read_csv('population.csv')
country = pd.read_csv('country_definitions.csv')
athlete = pd.read_csv('athlete_events.csv')

Now let's start cleaning the data

In [ ]:
#Let's see the missing data that we have
print(athlete.isnull().sum())

ID             0
Name           0
Sex            0
Age         9474
Height     60171
Weight     62875
Team           0
NOC            0
Games          0
Year           0
Season         0
City           0
Sport          0
Event          0
Medal     231333
dtype: int64


In [ ]:
#We are going to fill the NAN in the column Medals
athlete['Medal'].fillna('None', inplace=True)

In [ ]:
# Now we are going to see if the NOC is connected to only one team
print(athlete.loc[:, ['NOC', 'Team']].drop_duplicates()['NOC'].value_counts().head())

FRA    160
USA     97
GBR     96
SWE     52
NOR     46
Name: NOC, dtype: int64


In [ ]:
#See the information that we have in country
country

NOC       region                 notes
0    AFG  Afghanistan                   NaN
1    AHO      Curacao  Netherlands Antilles
2    ALB      Albania                   NaN
3    ALG      Algeria                   NaN
4    AND      Andorra                   NaN
..   ...          ...                   ...
225  YEM        Yemen                   NaN
226  YMD        Yemen           South Yemen
227  YUG       Serbia            Yugoslavia
228  ZAM       Zambia                   NaN
229  ZIM     Zimbabwe                   NaN

[230 rows x 3 columns]

### Now let's merge the files athletes and countries



In [ ]:
# merging the athelete file with the country
olympics= athlete.merge(country,
                                left_on = 'NOC',
                                right_on = 'NOC',
                                how = 'left')
olympics

ID                      Name Sex   Age  Height  Weight  \
0            1                 A Dijiang   M  24.0   180.0    80.0   
1            2                  A Lamusi   M  23.0   170.0    60.0   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0   
...        ...                       ...  ..   ...     ...     ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0   
271112  135570                  Piotr ya   M  27.0   176.0    59.0   
271113  135570                  Piotr ya   M  27.0   176.0    59.0   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0   

                  Team  NOC        Games  Year  Season            City  \
0                China  CHN  1992 Summer  1992  Summer       Barcelona   
1                China  CHN  2012 Summer  2012  Summer          London   
2              Denmark  DEN  1920 Summer  1920  Summer       Antwerpen   
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer           Paris   
4          Netherlands  NED  1988 Winter  1988  Winter         Calgary   
...                ...  ...          ...   ...     ...             ...   
271111        Poland-1  POL  1976 Winter  1976  Winter       Innsbruck   
271112          Poland  POL  2014 Winter  2014  Winter           Sochi   
271113          Poland  POL  2014 Winter  2014  Winter           Sochi   
271114          Poland  POL  1998 Winter  1998  Winter          Nagano   
271115          Poland  POL  2002 Winter  2002  Winter  Salt Lake City   

                Sport                                     Event Medal  \
0          Basketball               Basketball Men's Basketball  None   
1                Judo              Judo Men's Extra-Lightweight  None   
2            Football                   Football Men's Football  None   
3          Tug-Of-War               Tug-Of-War Men's Tug-Of-War  Gold   
4       Speed Skating          Speed Skating Women's 500 metres  None   
...               ...                                       ...   ...   
271111           Luge                Luge Mixed (Men)'s Doubles  None   
271112    Ski Jumping  Ski Jumping Men's Large Hill, Individual  None   
271113    Ski Jumping        Ski Jumping Men's Large Hill, Team  None   
271114      Bobsleigh                      Bobsleigh Men's Four  None   
271115      Bobsleigh                      Bobsleigh Men's Four  None   

             region notes  
0             China   NaN  
1             China   NaN  
2           Denmark   NaN  
3           Denmark   NaN  
4       Netherlands   NaN  
...             ...   ...  
271111       Poland   NaN  
271112       Poland   NaN  
271113       Poland   NaN  
271114       Poland   NaN  
271115       Poland   NaN  

[271116 rows x 17 columns]

In [ ]:
# See if we have NOC's that dont match in the column country
olympics.loc[olympics['region'].isnull(),['NOC', 'Team']].drop_duplicates()

NOC                      Team
578     SGP                 Singapore
6267    ROT  Refugee Olympic Athletes
44376   SGP              June Climene
61080   UNK                   Unknown
64674   TUV                    Tuvalu
80986   SGP                   Rika II
108582  SGP               Singapore-2
235895  SGP               Singapore-1

In [ ]:
# Replace missing Teams by the values above.
olympics['region'] = np.where(olympics['NOC']=='SGP', 'Singapore', olympics['region'])
olympics['region'] = np.where(olympics['NOC']=='ROT', 'Refugee Olympic Athletes', olympics['region'])
olympics['region'] = np.where(olympics['NOC']=='UNK', 'Unknown', olympics['region'])
olympics['region'] = np.where(olympics['NOC']=='TUV', 'Tuvalu', olympics['region'])


# Put these values from Country into Team
olympics.drop('Team', axis = 1, inplace = True)
olympics.rename(columns = {'region': 'Team'}, inplace = True)
olympics

ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season            City          Sport  \
0       1992 Summer  1992  Summer       Barcelona     Basketball   
1       2012 Summer  2012  Summer          London           Judo   
2       1920 Summer  1920  Summer       Antwerpen       Football   
3       1900 Summer  1900  Summer           Paris     Tug-Of-War   
4       1988 Winter  1988  Winter         Calgary  Speed Skating   
...             ...   ...     ...             ...            ...   
271111  1976 Winter  1976  Winter       Innsbruck           Luge   
271112  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271113  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271114  1998 Winter  1998  Winter          Nagano      Bobsleigh   
271115  2002 Winter  2002  Winter  Salt Lake City      Bobsleigh   

                                           Event Medal         Team notes  
0                    Basketball Men's Basketball  None        China   NaN  
1                   Judo Men's Extra-Lightweight  None        China   NaN  
2                        Football Men's Football  None      Denmark   NaN  
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark   NaN  
4               Speed Skating Women's 500 metres  None  Netherlands   NaN  
...                                          ...   ...          ...   ...  
271111                Luge Mixed (Men)'s Doubles  None       Poland   NaN  
271112  Ski Jumping Men's Large Hill, Individual  None       Poland   NaN  
271113        Ski Jumping Men's Large Hill, Team  None       Poland   NaN  
271114                      Bobsleigh Men's Four  None       Poland   NaN  
271115                      Bobsleigh Men's Four  None       Poland   NaN  

[271116 rows x 16 columns]

In [ ]:
#Dropping the column notes
olympics= olympics.drop('notes', axis=1)
olympics

ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season            City          Sport  \
0       1992 Summer  1992  Summer       Barcelona     Basketball   
1       2012 Summer  2012  Summer          London           Judo   
2       1920 Summer  1920  Summer       Antwerpen       Football   
3       1900 Summer  1900  Summer           Paris     Tug-Of-War   
4       1988 Winter  1988  Winter         Calgary  Speed Skating   
...             ...   ...     ...             ...            ...   
271111  1976 Winter  1976  Winter       Innsbruck           Luge   
271112  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271113  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271114  1998 Winter  1998  Winter          Nagano      Bobsleigh   
271115  2002 Winter  2002  Winter  Salt Lake City      Bobsleigh   

                                           Event Medal         Team  
0                    Basketball Men's Basketball  None        China  
1                   Judo Men's Extra-Lightweight  None        China  
2                        Football Men's Football  None      Denmark  
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark  
4               Speed Skating Women's 500 metres  None  Netherlands  
...                                          ...   ...          ...  
271111                Luge Mixed (Men)'s Doubles  None       Poland  
271112  Ski Jumping Men's Large Hill, Individual  None       Poland  
271113        Ski Jumping Men's Large Hill, Team  None       Poland  
271114                      Bobsleigh Men's Four  None       Poland  
271115                      Bobsleigh Men's Four  None       Poland  

[271116 rows x 15 columns]

### Now for the gdp file

In [ ]:
#Show the file gdp
gdp

Country Name Country Code  Year         Value
0       Arab World          ARB  1968  2.576068e+10
1       Arab World          ARB  1969  2.843420e+10
2       Arab World          ARB  1970  3.138550e+10
3       Arab World          ARB  1971  3.642691e+10
4       Arab World          ARB  1972  4.331606e+10
...            ...          ...   ...           ...
11502     Zimbabwe          ZWE  2012  1.424249e+10
11503     Zimbabwe          ZWE  2013  1.545177e+10
11504     Zimbabwe          ZWE  2014  1.589105e+10
11505     Zimbabwe          ZWE  2015  1.630467e+10
11506     Zimbabwe          ZWE  2016  1.661996e+10

[11507 rows x 4 columns]

In [ ]:
#Change the name of the columns Values and year
gdp.rename(columns={'Year':'Year','Value':'GDP'}, inplace=True)
gdp

Country Name Country Code  Year           GDP
0       Arab World          ARB  1968  2.576068e+10
1       Arab World          ARB  1969  2.843420e+10
2       Arab World          ARB  1970  3.138550e+10
3       Arab World          ARB  1971  3.642691e+10
4       Arab World          ARB  1972  4.331606e+10
...            ...          ...   ...           ...
11502     Zimbabwe          ZWE  2012  1.424249e+10
11503     Zimbabwe          ZWE  2013  1.545177e+10
11504     Zimbabwe          ZWE  2014  1.589105e+10
11505     Zimbabwe          ZWE  2015  1.630467e+10
11506     Zimbabwe          ZWE  2016  1.661996e+10

[11507 rows x 4 columns]

In [ ]:
# convert the year column to numeric
gdp['Year'] = pd.to_numeric(gdp['Year'], errors='coerce')
gdp

Country Name Country Code  Year           GDP
0       Arab World          ARB  1968  2.576068e+10
1       Arab World          ARB  1969  2.843420e+10
2       Arab World          ARB  1970  3.138550e+10
3       Arab World          ARB  1971  3.642691e+10
4       Arab World          ARB  1972  4.331606e+10
...            ...          ...   ...           ...
11502     Zimbabwe          ZWE  2012  1.424249e+10
11503     Zimbabwe          ZWE  2013  1.545177e+10
11504     Zimbabwe          ZWE  2014  1.589105e+10
11505     Zimbabwe          ZWE  2015  1.630467e+10
11506     Zimbabwe          ZWE  2016  1.661996e+10

[11507 rows x 4 columns]

Now we are going to see what is the best column to use for the joining of the two tables, the NOC and country code or NOC and country Name

In [ ]:
#Checking if NOCs in the olympics data match with those in the Country Code
len(list(set(olympics['NOC'].unique()) - set(gdp['Country Code'].unique())))

111

So, 111 NOCs in the olympics dataset have a representation in the gdp data country codes.

In [ ]:
#Lets try to use the name of the country is a better way to merge?
len(list(set(olympics['Team'].unique()) - set(gdp['Country Name'].unique())))

33

Since we only have 33 matched, so maybe what we can do is to add a country code for each Team in the olympics dataset first and then merge using the Country Code.

In [ ]:
# Merge the file gd with the olympics
olympics_ccode = olympics.merge(gdp[['Country Name', 'Country Code']].drop_duplicates(),
                                            left_on = 'Team',
                                            right_on = 'Country Name',
                                            how = 'left')

olympics_ccode.drop('Country Name', axis = 1, inplace = True)
olympics_ccode

ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season            City          Sport  \
0       1992 Summer  1992  Summer       Barcelona     Basketball   
1       2012 Summer  2012  Summer          London           Judo   
2       1920 Summer  1920  Summer       Antwerpen       Football   
3       1900 Summer  1900  Summer           Paris     Tug-Of-War   
4       1988 Winter  1988  Winter         Calgary  Speed Skating   
...             ...   ...     ...             ...            ...   
271111  1976 Winter  1976  Winter       Innsbruck           Luge   
271112  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271113  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271114  1998 Winter  1998  Winter          Nagano      Bobsleigh   
271115  2002 Winter  2002  Winter  Salt Lake City      Bobsleigh   

                                           Event Medal         Team  \
0                    Basketball Men's Basketball  None        China   
1                   Judo Men's Extra-Lightweight  None        China   
2                        Football Men's Football  None      Denmark   
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark   
4               Speed Skating Women's 500 metres  None  Netherlands   
...                                          ...   ...          ...   
271111                Luge Mixed (Men)'s Doubles  None       Poland   
271112  Ski Jumping Men's Large Hill, Individual  None       Poland   
271113        Ski Jumping Men's Large Hill, Team  None       Poland   
271114                      Bobsleigh Men's Four  None       Poland   
271115                      Bobsleigh Men's Four  None       Poland   

       Country Code  
0               CHN  
1               CHN  
2               DNK  
3               DNK  
4               NLD  
...             ...  
271111          POL  
271112          POL  
271113          POL  
271114          POL  
271115          POL  

[271116 rows x 16 columns]

In [ ]:
#Use the apply function to only update the 'Country Code' column in these specific case
olympics_ccode['Country Code'] = olympics_ccode.apply(lambda row: 'USA' if row['NOC'] == 'USA' else row['Country Code'], axis=1)
olympics_ccode

ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season            City          Sport  \
0       1992 Summer  1992  Summer       Barcelona     Basketball   
1       2012 Summer  2012  Summer          London           Judo   
2       1920 Summer  1920  Summer       Antwerpen       Football   
3       1900 Summer  1900  Summer           Paris     Tug-Of-War   
4       1988 Winter  1988  Winter         Calgary  Speed Skating   
...             ...   ...     ...             ...            ...   
271111  1976 Winter  1976  Winter       Innsbruck           Luge   
271112  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271113  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271114  1998 Winter  1998  Winter          Nagano      Bobsleigh   
271115  2002 Winter  2002  Winter  Salt Lake City      Bobsleigh   

                                           Event Medal         Team  \
0                    Basketball Men's Basketball  None        China   
1                   Judo Men's Extra-Lightweight  None        China   
2                        Football Men's Football  None      Denmark   
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark   
4               Speed Skating Women's 500 metres  None  Netherlands   
...                                          ...   ...          ...   
271111                Luge Mixed (Men)'s Doubles  None       Poland   
271112  Ski Jumping Men's Large Hill, Individual  None       Poland   
271113        Ski Jumping Men's Large Hill, Team  None       Poland   
271114                      Bobsleigh Men's Four  None       Poland   
271115                      Bobsleigh Men's Four  None       Poland   

       Country Code  
0               CHN  
1               CHN  
2               DNK  
3               DNK  
4               NLD  
...             ...  
271111          POL  
271112          POL  
271113          POL  
271114          POL  
271115          POL  

[271116 rows x 16 columns]

In [ ]:
# Merge the gdp and the olympics_code
olympics_gdp = olympics_ccode.merge(gdp, left_on=['NOC', 'Year'], right_on=['Country Code', 'Year'], how='left')
olympics_gdp.drop(['Country Name','Country Code_y'], axis = 1, inplace = True)
olympics_gdp.rename(columns = {'Country Code_x': 'Country Code'}, inplace = True)
olympics_gdp


ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season            City          Sport  \
0       1992 Summer  1992  Summer       Barcelona     Basketball   
1       2012 Summer  2012  Summer          London           Judo   
2       1920 Summer  1920  Summer       Antwerpen       Football   
3       1900 Summer  1900  Summer           Paris     Tug-Of-War   
4       1988 Winter  1988  Winter         Calgary  Speed Skating   
...             ...   ...     ...             ...            ...   
271111  1976 Winter  1976  Winter       Innsbruck           Luge   
271112  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271113  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271114  1998 Winter  1998  Winter          Nagano      Bobsleigh   
271115  2002 Winter  2002  Winter  Salt Lake City      Bobsleigh   

                                           Event Medal         Team  \
0                    Basketball Men's Basketball  None        China   
1                   Judo Men's Extra-Lightweight  None        China   
2                        Football Men's Football  None      Denmark   
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark   
4               Speed Skating Women's 500 metres  None  Netherlands   
...                                          ...   ...          ...   
271111                Luge Mixed (Men)'s Doubles  None       Poland   
271112  Ski Jumping Men's Large Hill, Individual  None       Poland   
271113        Ski Jumping Men's Large Hill, Team  None       Poland   
271114                      Bobsleigh Men's Four  None       Poland   
271115                      Bobsleigh Men's Four  None       Poland   

       Country Code           GDP  
0               CHN  4.269157e+11  
1               CHN  8.560547e+12  
2               DNK           NaN  
3               DNK           NaN  
4               NLD           NaN  
...             ...           ...  
271111          POL           NaN  
271112          POL  5.450759e+11  
271113          POL  5.450759e+11  
271114          POL  1.743883e+11  
271115          POL  1.986806e+11  

[271116 rows x 17 columns]

# Now lets do for the population set

In [ ]:
#See the file population
population

geo         name  time  Population
0      afg  Afghanistan  1800   3280000.0
1      afg  Afghanistan  1801   3280000.0
2      afg  Afghanistan  1802   3280000.0
3      afg  Afghanistan  1803   3280000.0
4      afg  Afghanistan  1804   3280000.0
...    ...          ...   ...         ...
59292  zwe     Zimbabwe  2096  35469605.0
59293  zwe     Zimbabwe  2097  35517791.0
59294  zwe     Zimbabwe  2098  35561279.0
59295  zwe     Zimbabwe  2099  35599779.0
59296  zwe     Zimbabwe  2100  35633415.0

[59297 rows x 4 columns]

In [ ]:
#Drop all the rows that are all blanks
population = population.dropna(how='all')
population

geo         name  time  Population
0      afg  Afghanistan  1800   3280000.0
1      afg  Afghanistan  1801   3280000.0
2      afg  Afghanistan  1802   3280000.0
3      afg  Afghanistan  1803   3280000.0
4      afg  Afghanistan  1804   3280000.0
...    ...          ...   ...         ...
59292  zwe     Zimbabwe  2096  35469605.0
59293  zwe     Zimbabwe  2097  35517791.0
59294  zwe     Zimbabwe  2098  35561279.0
59295  zwe     Zimbabwe  2099  35599779.0
59296  zwe     Zimbabwe  2100  35633415.0

[59297 rows x 4 columns]

In [ ]:
#Transform the column geo into upper letters
population['geo'] = population['geo'].str.upper()
population

geo         name  time  Population
0      AFG  Afghanistan  1800   3280000.0
1      AFG  Afghanistan  1801   3280000.0
2      AFG  Afghanistan  1802   3280000.0
3      AFG  Afghanistan  1803   3280000.0
4      AFG  Afghanistan  1804   3280000.0
...    ...          ...   ...         ...
59292  ZWE     Zimbabwe  2096  35469605.0
59293  ZWE     Zimbabwe  2097  35517791.0
59294  ZWE     Zimbabwe  2098  35561279.0
59295  ZWE     Zimbabwe  2099  35599779.0
59296  ZWE     Zimbabwe  2100  35633415.0

[59297 rows x 4 columns]

In [ ]:
#Checking if NOCs in the olympics data match with those in the Country Code
len(list(set(olympics_gdp['NOC'].unique()) - set(population['geo'].unique())))

110

In [ ]:
#Lets try to use the name of the country is a better way to merge?
len(list(set(olympics_gdp['Team'].unique()) - set(population['name'].unique())))

30

In [ ]:
# Merge the olympics_gdp and the population
olympics_complete = olympics_gdp.merge(
    population,
    left_on=['Country Code', 'Year'],
    right_on=['geo', 'time'],
    how='left'
)
olympics_complete

ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season  ...          Sport  \
0       1992 Summer  1992  Summer  ...     Basketball   
1       2012 Summer  2012  Summer  ...           Judo   
2       1920 Summer  1920  Summer  ...       Football   
3       1900 Summer  1900  Summer  ...     Tug-Of-War   
4       1988 Winter  1988  Winter  ...  Speed Skating   
...             ...   ...     ...  ...            ...   
271111  1976 Winter  1976  Winter  ...           Luge   
271112  2014 Winter  2014  Winter  ...    Ski Jumping   
271113  2014 Winter  2014  Winter  ...    Ski Jumping   
271114  1998 Winter  1998  Winter  ...      Bobsleigh   
271115  2002 Winter  2002  Winter  ...      Bobsleigh   

                                           Event Medal         Team  \
0                    Basketball Men's Basketball  None        China   
1                   Judo Men's Extra-Lightweight  None        China   
2                        Football Men's Football  None      Denmark   
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark   
4               Speed Skating Women's 500 metres  None  Netherlands   
...                                          ...   ...          ...   
271111                Luge Mixed (Men)'s Doubles  None       Poland   
271112  Ski Jumping Men's Large Hill, Individual  None       Poland   
271113        Ski Jumping Men's Large Hill, Team  None       Poland   
271114                      Bobsleigh Men's Four  None       Poland   
271115                      Bobsleigh Men's Four  None       Poland   

       Country Code           GDP  geo         name    time    Population  
0               CHN  4.269157e+11  CHN        China  1992.0  1.183813e+09  
1               CHN  8.560547e+12  CHN        China  2012.0  1.366561e+09  
2               DNK           NaN  DNK      Denmark  1920.0  3.241362e+06  
3               DNK           NaN  DNK      Denmark  1900.0  2.565333e+06  
4               NLD           NaN  NLD  Netherlands  1988.0  1.476232e+07  
...             ...           ...  ...          ...     ...           ...  
271111          POL           NaN  POL       Poland  1976.0  3.422971e+07  
271112          POL  5.450759e+11  POL       Poland  2014.0  3.858187e+07  
271113          POL  5.450759e+11  POL       Poland  2014.0  3.858187e+07  
271114          POL  1.743883e+11  POL       Poland  1998.0  3.837999e+07  
271115          POL  1.986806e+11  POL       Poland  2002.0  3.864748e+07  

[271116 rows x 21 columns]

In [ ]:
#Drop the columns geo and time
olympics_complete.drop(['geo','time'], axis = 1, inplace = True)
olympics_complete

ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season            City          Sport  \
0       1992 Summer  1992  Summer       Barcelona     Basketball   
1       2012 Summer  2012  Summer          London           Judo   
2       1920 Summer  1920  Summer       Antwerpen       Football   
3       1900 Summer  1900  Summer           Paris     Tug-Of-War   
4       1988 Winter  1988  Winter         Calgary  Speed Skating   
...             ...   ...     ...             ...            ...   
271111  1976 Winter  1976  Winter       Innsbruck           Luge   
271112  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271113  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271114  1998 Winter  1998  Winter          Nagano      Bobsleigh   
271115  2002 Winter  2002  Winter  Salt Lake City      Bobsleigh   

                                           Event Medal         Team  \
0                    Basketball Men's Basketball  None        China   
1                   Judo Men's Extra-Lightweight  None        China   
2                        Football Men's Football  None      Denmark   
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark   
4               Speed Skating Women's 500 metres  None  Netherlands   
...                                          ...   ...          ...   
271111                Luge Mixed (Men)'s Doubles  None       Poland   
271112  Ski Jumping Men's Large Hill, Individual  None       Poland   
271113        Ski Jumping Men's Large Hill, Team  None       Poland   
271114                      Bobsleigh Men's Four  None       Poland   
271115                      Bobsleigh Men's Four  None       Poland   

       Country Code           GDP         name    Population  
0               CHN  4.269157e+11        China  1.183813e+09  
1               CHN  8.560547e+12        China  1.366561e+09  
2               DNK           NaN      Denmark  3.241362e+06  
3               DNK           NaN      Denmark  2.565333e+06  
4               NLD           NaN  Netherlands  1.476232e+07  
...             ...           ...          ...           ...  
271111          POL           NaN       Poland  3.422971e+07  
271112          POL  5.450759e+11       Poland  3.858187e+07  
271113          POL  5.450759e+11       Poland  3.858187e+07  
271114          POL  1.743883e+11       Poland  3.837999e+07  
271115          POL  1.986806e+11       Poland  3.864748e+07  

[271116 rows x 19 columns]

## Now that we have all the data in one data set, we are going to arrange the information

In [ ]:
#Create a column that say 1 if they win a medal and 0 if they dont win a medal
olympics_complete['Medal_Won'] = np.where(olympics_complete.loc[:,'Medal'] == 'None', 0, 1)
olympics_complete

ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season            City          Sport  \
0       1992 Summer  1992  Summer       Barcelona     Basketball   
1       2012 Summer  2012  Summer          London           Judo   
2       1920 Summer  1920  Summer       Antwerpen       Football   
3       1900 Summer  1900  Summer           Paris     Tug-Of-War   
4       1988 Winter  1988  Winter         Calgary  Speed Skating   
...             ...   ...     ...             ...            ...   
271111  1976 Winter  1976  Winter       Innsbruck           Luge   
271112  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271113  2014 Winter  2014  Winter           Sochi    Ski Jumping   
271114  1998 Winter  1998  Winter          Nagano      Bobsleigh   
271115  2002 Winter  2002  Winter  Salt Lake City      Bobsleigh   

                                           Event Medal         Team  \
0                    Basketball Men's Basketball  None        China   
1                   Judo Men's Extra-Lightweight  None        China   
2                        Football Men's Football  None      Denmark   
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark   
4               Speed Skating Women's 500 metres  None  Netherlands   
...                                          ...   ...          ...   
271111                Luge Mixed (Men)'s Doubles  None       Poland   
271112  Ski Jumping Men's Large Hill, Individual  None       Poland   
271113        Ski Jumping Men's Large Hill, Team  None       Poland   
271114                      Bobsleigh Men's Four  None       Poland   
271115                      Bobsleigh Men's Four  None       Poland   

       Country Code           GDP         name    Population  Medal_Won  
0               CHN  4.269157e+11        China  1.183813e+09          0  
1               CHN  8.560547e+12        China  1.366561e+09          0  
2               DNK           NaN      Denmark  3.241362e+06          0  
3               DNK           NaN      Denmark  2.565333e+06          1  
4               NLD           NaN  Netherlands  1.476232e+07          0  
...             ...           ...          ...           ...        ...  
271111          POL           NaN       Poland  3.422971e+07          0  
271112          POL  5.450759e+11       Poland  3.858187e+07          0  
271113          POL  5.450759e+11       Poland  3.858187e+07          0  
271114          POL  1.743883e+11       Poland  3.837999e+07          0  
271115          POL  1.986806e+11       Poland  3.864748e+07          0  

[271116 rows x 20 columns]

In [ ]:
#Identify team events and update 'Team Sports' column to 'yes'
identify_team_events = pd.pivot_table(olympics_complete,
                                       index=['Team', 'Year', 'Event'],
                                       columns='Medal',
                                       values='Medal_Won',
                                       aggfunc='sum',
                                       fill_value=0).drop('None', axis=1).reset_index()

identify_team_events = identify_team_events.loc[identify_team_events['Gold'] > 1, :]

team_sports = identify_team_events['Event'].unique()

#Update 'Team Sports' column in the original datset
olympics_complete.loc[olympics_complete['Event'].isin(team_sports), 'Team Sports'] = 'Yes'
olympics_complete

ID                      Name Sex   Age  Height  Weight  NOC  \
0            1                 A Dijiang   M  24.0   180.0    80.0  CHN   
1            2                  A Lamusi   M  23.0   170.0    60.0  CHN   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED   
...        ...                       ...  ..   ...     ...     ...  ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0  POL   
271112  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271113  135570                  Piotr ya   M  27.0   176.0    59.0  POL   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0  POL   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0  POL   

              Games  Year  Season  ...          Sport  \
0       1992 Summer  1992  Summer  ...     Basketball   
1       2012 Summer  2012  Summer  ...           Judo   
2       1920 Summer  1920  Summer  ...       Football   
3       1900 Summer  1900  Summer  ...     Tug-Of-War   
4       1988 Winter  1988  Winter  ...  Speed Skating   
...             ...   ...     ...  ...            ...   
271111  1976 Winter  1976  Winter  ...           Luge   
271112  2014 Winter  2014  Winter  ...    Ski Jumping   
271113  2014 Winter  2014  Winter  ...    Ski Jumping   
271114  1998 Winter  1998  Winter  ...      Bobsleigh   
271115  2002 Winter  2002  Winter  ...      Bobsleigh   

                                           Event Medal         Team  \
0                    Basketball Men's Basketball  None        China   
1                   Judo Men's Extra-Lightweight  None        China   
2                        Football Men's Football  None      Denmark   
3                    Tug-Of-War Men's Tug-Of-War  Gold      Denmark   
4               Speed Skating Women's 500 metres  None  Netherlands   
...                                          ...   ...          ...   
271111                Luge Mixed (Men)'s Doubles  None       Poland   
271112  Ski Jumping Men's Large Hill, Individual  None       Poland   
271113        Ski Jumping Men's Large Hill, Team  None       Poland   
271114                      Bobsleigh Men's Four  None       Poland   
271115                      Bobsleigh Men's Four  None       Poland   

       Country Code           GDP         name    Population  Medal_Won  \
0               CHN  4.269157e+11        China  1.183813e+09          0   
1               CHN  8.560547e+12        China  1.366561e+09          0   
2               DNK           NaN      Denmark  3.241362e+06          0   
3               DNK           NaN      Denmark  2.565333e+06          1   
4               NLD           NaN  Netherlands  1.476232e+07          0   
...             ...           ...          ...           ...        ...   
271111          POL           NaN       Poland  3.422971e+07          0   
271112          POL  5.450759e+11       Poland  3.858187e+07          0   
271113          POL  5.450759e+11       Poland  3.858187e+07          0   
271114          POL  1.743883e+11       Poland  3.837999e+07          0   
271115          POL  1.986806e+11       Poland  3.864748e+07          0   

        Team Sports  
0               Yes  
1               NaN  
2               Yes  
3               Yes  
4               NaN  
...             ...  
271111          Yes  
271112          NaN  
271113          Yes  
271114          Yes  
271115          Yes  

[271116 rows x 21 columns]

In [ ]:
#Remove the individual sports identified in the team_sports
remove_sports = ["Gymnastics Women's Balance Beam", "Gymnastics Men's Horizontal Bar",
                 "Swimming Women's 100 metres Freestyle", "Swimming Men's 50 metres Freestyle","Alpinism Mixed Alpinism","Art Competitions Mixed Architecture","Art Competitions Mixed Architecture, Architectural Designs","Art Competitions Mixed Architecture", "Designs For Town Planning","Motorboating Mixed C-Class","Speed Skating Men's 1,500 metres","Athletics MenAthletics Mens Pole Vaults Pole Vault","Cross Country Skiing Mens 10/10 kilometres Pursuit","Figure Skating Mixed Ice Dancing"]

team_sports = list(set(team_sports) - set(remove_sports))

In [ ]:
#Download the file
olympics_complete.to_csv("olympics_complete.csv", index=False)